---
title: "MP_SRF_miTags_18S_and_16S_diversity_analysis"
author: "lrubinat"
date: "03/05/2016"
output:
  html_document:
    theme: united
    toc: yes
  pdf_document:
    highlight: zenburn
    toc: yes
---

<!--- INITIALIZATION
```{r, echo=FALSE}
#error hook to kill knitr in case of errors
library(knitr)
knit_hooks$set(error = function(x, options) stop(x))
opts_chunk$set(cache=TRUE, autodep=TRUE)
```
--->

# 1) 18S amplicons

## 1.1) Data overview

Let's read the dataset and remove the samples containing less than 470 reads:

``` {r load_data_18S, echo=FALSE, message=FALSE}
setwd("/home/laura/Documents/TFM/genwork/data_analysis/MP_SRF_miTags_18S_and_16S_diversity_analysis/")

#read data 
MP_18S_miTags_tax <- read.table(file="/home/laura/Documents/TFM/home/data/TARASPINA/SRF_miTags/MP_SRF/otu_table99_MP_SRF_18S.txt", head=TRUE, fill=TRUE)

#table dimensions and format before setting column names
dim(MP_18S_miTags_tax) # 1641   13
MP_18S_miTags_tax[1:5,1:5]

#row names = OTU name (option A)
row.names(MP_18S_miTags_tax)<-MP_18S_miTags_tax[,1]

#row names = row number (option B)
#rownames(otu_MP_18S) <- 1:nrow(otu_MP_18S)

MP_18S_miTags_tax<-MP_18S_miTags_tax[,-1]
MP_18S_miTags_tax[is.na(MP_18S_miTags_tax)]<-"NA"

dim(MP_18S_miTags_tax)
MP_18S_miTags_tax[1:5,1:5]

#table with taxonomi classification alone
MP_18S_class <- MP_18S_miTags_tax[,11:12]
dim(MP_18S_class)
MP_18S_class[1:5,1:2]

#table with occurence data alone
MP_18S_miTags_tax_occur <- MP_18S_miTags_tax[,1:10]
dim(MP_18S_miTags_tax_occur) #1641   10
MP_18S_miTags_tax_occur[1:5,1:5]

amplicons_per_sample_MP_18S<-colSums(MP_18S_miTags_tax_occur)
amplicons_per_sample_MP_18S[which(colSums(MP_18S_miTags_tax_occur)<470)]
# sample st030 has less than 470 reads.

#remove samples with less than 470 reads
MP_18S_miTags_tax_occur_min470 <- MP_18S_miTags_tax_occur[,colSums(MP_18S_miTags_tax_occur) >= 470]
dim(MP_18S_miTags_tax_occur_min470)
```


## 1.2) Normalization

Let's normalize the original dataset by randomly subsampling 470 reads in each station:

``` {r species_richness_rarefaction1_18S, echo=TRUE}
library(vegan)
MP_18S_miTags_tax_occur_min470_t<-t(MP_18S_miTags_tax_occur_min470)
MP_18S_miTags_tax_occur_ss470<-rrarefy(MP_18S_miTags_tax_occur_min470_t, 470)
```

The normalized table shows the following dimensions and format:

```{r species_richness_rarefaction2_18S, echo=FALSE}
dim(MP_18S_miTags_tax_occur_ss470)
MP_18S_miTags_tax_occur_ss470[1:5,1:5]
```

Its content fits with the expected normalization values (470 reads per station):

``` {r species_richness_rarefaction3_18S, echo=TRUE}
rowSums(MP_18S_miTags_tax_occur_ss470)
```

Let's check out how many OTUs don't appear in the new table:

```{r species_richness_rarefaction4_18S, echo=1:5}
length(which(colSums(MP_18S_miTags_tax_occur_ss470)==0)) 
```

There are 429 OTUs that don't show any occurrence in the normalized data. Let's remove them from the table and take a look at its final dimensions:

```{r species_richness_rarefaction5_18S, echo=1:3}
MP_18S_miTags_tax_occur_ss470_no_cero<-MP_18S_miTags_tax_occur_ss470[,-(which(colSums(MP_18S_miTags_tax_occur_ss470)==0))]
dim(MP_18S_miTags_tax_occur_ss470_no_cero)

#the final dimensions of the normalized table are 119 26907.
#1212 + 429 = 1641
```

Datasets summary:
dim(MP_18S_miTags_tax) --> 1641   12
dim(MP_18S_miTags_tax_occur) --> 1641   10
dim(MP_18S_miTags_tax_occur_ss470_no_cero) --> 9 1212


## 1.3) Taxonomic composition analysis

Let's add the taxonomic classification to the left OTUs by merging "MP_18S_miTags_tax_occur_ss470_no_cero" with "MP_18S_tax":

```{r merge_tables_18S, echo=FALSE}
MP_18S_miTags_tax_occur_ss470_no_cero_t<-t(MP_18S_miTags_tax_occur_ss470_no_cero)
MP_18S_miTags_tax<-merge(MP_18S_miTags_tax_occur_ss470_no_cero_t,MP_18S_class, by="row.names")

dim(MP_18S_miTags_tax)
MP_18S_miTags_tax[1:5,1:5]

#fix OTU_no as new row
rownames(MP_18S_miTags_tax)=MP_18S_miTags_tax$Row.names

#add OTU_no as rowname
rownames.MP_18S_miTags_tax<-MP_18S_miTags_tax[,1]
MP_18S_miTags_tax<-MP_18S_miTags_tax[,-1]
#colnames(MP_18S_miTags_tax, do.NULL=F)

dim(MP_18S_miTags_tax)
MP_18S_miTags_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
MP_18S_miTags_tax["OTU_no"] <- NA
MP_18S_miTags_tax$OTU_no <- sapply(strsplit(rownames(MP_18S_miTags_tax),split= "\\_"),'[',2)
MP_18S_miTags_tax$OTU_no <- as.numeric(as.character(MP_18S_miTags_tax$OTU_no))
MP_18S_miTags_tax_sorted<-MP_18S_miTags_tax[order(MP_18S_miTags_tax$OTU_no, decreasing = FALSE), ]

dim(MP_18S_miTags_tax_sorted)
MP_18S_miTags_tax_sorted[1:5,1:5]
```

```{r select_phototrophs_18S, echo=FALSE} 
MP_18S_miTags_phototrophs <- MP_18S_miTags_tax_sorted[which(MP_18S_miTags_tax_sorted$class != "heterotrophic_protist"),]

dim(MP_18S_miTags_phototrophs)
```

```{r count_samples_per_group_18S, echo=FALSE}
#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Dinophyceae"),]
Dinophyceae_tb[1:5,1:5]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:9]
Dinophyceae_tb_occur[1:5,1:5]
dim(Dinophyceae_tb_occur)

length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
#Dinophyceae_tb_samples <- Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0]
#length(Dinophyceae_tb_samples[which(colSums(Dinophyceae_tb_occur) != 0)])

Prasinophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:9]
length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])

Chrysophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:9]
length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])

Pelagophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:9]
length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])

Dictyochophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:9]
length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])

Cryptomonadales_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Cryptomonadales"),]
Cryptomonadales_tb_occur <- Cryptomonadales_tb[,1:9]
length(Cryptomonadales_tb_occur[,colSums(Cryptomonadales_tb_occur) > 0])

Bacillariophyta_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Bacillariophyta"),]
Bacillariophyta_tb_occur <- Bacillariophyta_tb[,1:9]
length(Bacillariophyta_tb_occur[,colSums(Bacillariophyta_tb_occur) > 0])

Chlorarachniophyta_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Chlorarachniophyta"),]
Chlorarachniophyta_tb_occur <- Chlorarachniophyta_tb[,1:9]
length(Chlorarachniophyta_tb_occur[,colSums(Chlorarachniophyta_tb_occur) > 0])

Bolidophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:9]
length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])

Pinguiochysidales_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Pinguiochysidales"),]
Pinguiochysidales_tb_occur <- Pinguiochysidales_tb[,1:9]
length(Pinguiochysidales_tb_occur[,colSums(Pinguiochysidales_tb_occur) > 0])

Prymnesiophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:9]
length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])

Mamiellophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:9]
length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])

Eustigmatales_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Eustigmatales"),]
Eustigmatales_tb_occur <- Eustigmatales_tb[,1:9]
length(Eustigmatales_tb_occur[,colSums(Eustigmatales_tb_occur) > 0])

Chlorophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:9]
length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])

Ulvophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:9]
length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])

Raphydophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:9]
length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])

Trebouxiophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:9]
length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])

Phaeothamniophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:9]
length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])

Xanthophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:9]
length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])

Chlorodendrophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:9]
length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])

Nephroselmis_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Nephroselmis"),]
Nephroselmis_tb_occur <- Nephroselmis_tb[,1:9]
length(Nephroselmis_tb_occur[,colSums(Nephroselmis_tb_occur) > 0])

Rhodophyta_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Rhodophyta"),]
Rhodophyta_tb_occur <- Rhodophyta_tb[,1:9]
length(Rhodophyta_tb_occur[,colSums(Rhodophyta_tb_occur) > 0])

Pavlovophyceae_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:9]
length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])

Rappemonad_tb <- MP_18S_miTags_phototrophs[which(MP_18S_miTags_phototrophs$class == "Rappemonad"),]
Rappemonad_tb_occur <- Rappemonad_tb[,1:9]
length(Rappemonad_tb_occur[,colSums(Rappemonad_tb_occur) > 0])
```

```{r aggregate_18S, echo=FALSE}
MP_18S_miTags_class_summary_reads_per_class<-aggregate(rowSums(MP_18S_miTags_phototrophs[1:9]), list(MP_18S_miTags_phototrophs$class), sum)
# count the different groups

MP_18S_miTags_class_summary_otus_per_class<-aggregate(rowSums(MP_18S_miTags_phototrophs[1:9]), list(MP_18S_miTags_phototrophs$class), length)

## READS PER CLASS ##
attach(MP_18S_miTags_class_summary_reads_per_class)
MP_18S_miTags_class_summary_reads_per_class_order<-MP_18S_miTags_class_summary_reads_per_class[order(-x),]
detach(MP_18S_miTags_class_summary_reads_per_class)
MP_18S_miTags_class_summary_reads_per_class_order

#fix column names
row.names(MP_18S_miTags_class_summary_reads_per_class_order)<-MP_18S_miTags_class_summary_reads_per_class_order[,1]
MP_18S_miTags_class_summary_reads_per_class_order<-MP_18S_miTags_class_summary_reads_per_class_order[c(-1)]
colnames(MP_18S_miTags_class_summary_reads_per_class_order)<-c("reads_per_class")

## OTUs PER CLASS ##
attach(MP_18S_miTags_class_summary_otus_per_class)
MP_18S_miTags_class_summary_otus_per_class_order<-MP_18S_miTags_class_summary_otus_per_class[order(-x),]
detach(MP_18S_miTags_class_summary_otus_per_class)
MP_18S_miTags_class_summary_otus_per_class_order

row.names(MP_18S_miTags_class_summary_otus_per_class_order)<-MP_18S_miTags_class_summary_otus_per_class_order[,1]
MP_18S_miTags_class_summary_otus_per_class_order<-MP_18S_miTags_class_summary_otus_per_class_order[c(-1)]
colnames(MP_18S_miTags_class_summary_otus_per_class_order)<-c("OTUs_per_class")

#class_summary_reads<-aggregate(sum~class, data=otutab_full_wTax, FUN="sum") 
# sum reads different groups
```

```{r create_table_merging_#OTUs_#reads_#samples_18S, echo=FALSE}
MP_18S_miTags_OTUs_reads_samples <- merge(MP_18S_miTags_class_summary_reads_per_class_order, MP_18S_miTags_class_summary_otus_per_class_order, by="row.names")

row.names(MP_18S_miTags_OTUs_reads_samples)<-MP_18S_miTags_OTUs_reads_samples[,1]
MP_18S_miTags_OTUs_reads_samples<-MP_18S_miTags_OTUs_reads_samples[,-1]
colnames(MP_18S_miTags_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
MP_18S_miTags_OTUs_reads_samples[,1:2]

MP_18S_miTags_OTUs_reads_samples<-MP_18S_miTags_OTUs_reads_samples[order(MP_18S_miTags_OTUs_reads_samples$reads_per_class, MP_18S_miTags_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
MP_18S_miTags_OTUs_reads_samples["samples_per_class"] <- NA
MP_18S_miTags_OTUs_reads_samples$samples_per_class<-c(9,9,7,9,9,9,7,8,6,8,4,2,2,1,1,1)
MP_18S_miTags_OTUs_reads_samples
```

```{r stations_vs_OTUs_18S, echo=FALSE}
library(ggplot2)

#p <- ggplot(MP_18S_miTags_OTUs_reads_samples, aes(samples_per_class,OTUs_per_class))
#p + geom_point()
MP_18S_miTags_otus_vs_samples_plot <- ggplot(MP_18S_miTags_OTUs_reads_samples, aes(samples_per_class,OTUs_per_class, label = rownames(MP_18S_miTags_OTUs_reads_samples)))
MP_18S_miTags_otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 11)) + labs(title="[tb1] #samples vs #OTUs", x="samples", y="OTUs")
```

```{r OTUs_vs_reads_18S, echo=FALSE}
MP_18S_miTags_otus_vs_samples_plot <- ggplot(MP_18S_miTags_OTUs_reads_samples, aes(OTUs_per_class, reads_per_class, label = rownames(MP_18S_miTags_OTUs_reads_samples)))
MP_18S_miTags_otus_vs_samples_plot + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size = 3) + scale_x_continuous(limits = c(0, 150)) + labs(title="[tb3] #OTUs vs #reads", x="OTUs", y="reads") + coord_trans(y = "log10")
```


# 2) 16S amplicons

## 2.1) Data overview

Let's read the dataset and remove the samples containing less than 470 reads:

``` {r load_data_16S, echo=FALSE, message=FALSE}

#read data 
MP_16S_miTags_tax <- read.table(file="/home/laura/Documents/TFM/home/data/TARASPINA/SRF_miTags/MP_SRF/otu_table99_MP_SRF_16S.txt", head=TRUE, fill=TRUE)

#table dimensions and format before setting column names
dim(MP_16S_miTags_tax) # 8248   14
MP_16S_miTags_tax[1:5,1:5]

#row names = OTU name (option A)
row.names(MP_16S_miTags_tax)<-MP_16S_miTags_tax[,1]

#row names = row number (option B)
#rownames(otu_MP_16S) <- 1:nrow(otu_MP_16S)

MP_16S_miTags_tax<-MP_16S_miTags_tax[,-1]
MP_16S_miTags_tax[is.na(MP_16S_miTags_tax)]<-"NA"

dim(MP_16S_miTags_tax)
MP_16S_miTags_tax[1:5,1:5]

#table with taxonomi classification alone
MP_16S_miTags_class <- MP_16S_miTags_tax[,11:13]
dim(MP_16S_miTags_class)
MP_16S_miTags_class[1:5,1:3]

#table with occurence data alone
MP_16S_miTags_tax_occur <- MP_16S_miTags_tax[,1:10]
dim(MP_16S_miTags_tax_occur) #8248   10
MP_16S_miTags_tax_occur[1:5,1:5]

amplicons_per_sample_MP_16S<-colSums(MP_16S_miTags_tax_occur)
amplicons_per_sample_MP_16S[which(colSums(MP_16S_miTags_tax_occur)<20000)]
#only st101 and st120 have less than 20000 ready. We'll normalize at 13676 reads.

#let's remove samples with omitted in 18S dataset (so that we can compare the relative abundance of 18S and 16S OTUs considering the same samples)
MP_16S_miTags_tax_occur_min13676<-subset(MP_16S_miTags_tax_occur, select=-c(st030))
dim(MP_16S_miTags_tax_occur_min13676) #8248    9
MP_16S_miTags_tax_occur_min13676[1:5,1:5]
```

## 2.2) Normalization

Let's normalize the original dataset by randomly subsampling 13676 reads in each station:

``` {r species_richness_rarefaction1_16S, echo=TRUE}
MP_16S_miTags_tax_occur_min13676_t<-t(MP_16S_miTags_tax_occur_min13676)
MP_16S_miTags_tax_occur_ss13676<-rrarefy(MP_16S_miTags_tax_occur_min13676_t, 13676)
```

The normalized table shows the following dimensions and format:

```{r species_richness_rarefaction2_16S, echo=FALSE}
dim(MP_16S_miTags_tax_occur_ss13676)
MP_16S_miTags_tax_occur_ss13676[1:5,1:5]
```

Its content fits with the expected normalization values (13676 reads per station):

``` {r species_richness_rarefaction3_16S, echo=TRUE}
rowSums(MP_16S_miTags_tax_occur_ss13676)
```

Let's check out how many OTUs don't appear in the new table:

```{r species_richness_rarefaction4_16S, echo=1:5}
length(which(colSums(MP_16S_miTags_tax_occur_ss13676)==0)) 
```

There are 2174 OTUs that don't show any occurrence in the normalized data. Let's remove them from the table and take a look at its final dimensions:

```{r species_richness_rarefaction5_16S, echo=1:3}
MP_16S_miTags_tax_occur_ss13676_no_cero<-MP_16S_miTags_tax_occur_ss13676[,-(which(colSums(MP_16S_miTags_tax_occur_ss13676)==0))]
dim(MP_16S_miTags_tax_occur_ss13676_no_cero)

#the final dimensions of the normalized table are  9 6074.
#2174+6074 = 8248
```

Datasets summary:
dim(MP_16S_miTags_tax) --> 8248   13
dim(MP_16S_miTags_tax_occur) --> 8248   10
dim(MP_16S_miTags_tax_occur_ss13676_no_cero) -->  9 6074

## 2.3) Taxonomic composition analysis

Let's add the taxonomic classification to the left OTUs by merging "MP_16S_miTags_tax_occur_ss13676_no_cero" with "MP_16S_miTags_tax".

```{r merge_tables_16S, echo=FALSE}
MP_16S_miTags_tax_occur_ss13676_no_cero_t<-t(MP_16S_miTags_tax_occur_ss13676_no_cero)
MP_16S_miTags_ss13676_tax<-merge(MP_16S_miTags_tax_occur_ss13676_no_cero_t,MP_16S_miTags_class, by="row.names")

dim(MP_16S_miTags_ss13676_tax)
MP_16S_miTags_ss13676_tax[1:5,1:5]

#fix OTU_no as new row
rownames(MP_16S_miTags_ss13676_tax)=MP_16S_miTags_ss13676_tax$Row.names

#add OTU_no as rowname
rownames.MP_16S_miTags_ss13676_tax<-MP_16S_miTags_ss13676_tax[,1]
MP_16S_miTags_ss13676_tax<-MP_16S_miTags_ss13676_tax[,-1]
#colnames(MP_16S_miTags_ss13676_tax, do.NULL=F)

dim(MP_16S_miTags_ss13676_tax)
MP_16S_miTags_ss13676_tax[1:5, 1:5]

#sort by OTU_no (split rowname, introduce no. into column "OTU_no" and sort)
MP_16S_miTags_ss13676_tax["OTU_no"] <- NA
MP_16S_miTags_ss13676_tax$OTU_no <- sapply(strsplit(rownames(MP_16S_miTags_ss13676_tax),split= "\\_"),'[',2)
MP_16S_miTags_ss13676_tax$OTU_no <- as.numeric(as.character(MP_16S_miTags_ss13676_tax$OTU_no))
MP_16S_miTags_ss13676_tax_sorted<-MP_16S_miTags_ss13676_tax[order(MP_16S_miTags_ss13676_tax$OTU_no, decreasing = FALSE), ]

dim(MP_16S_miTags_ss13676_tax_sorted)
MP_16S_miTags_ss13676_tax_sorted[1:5,10:11]
```

```{r select_phototrophs_16S2, echo=FALSE}
MP_16S_miTags_bacteria <- MP_16S_miTags_ss13676_tax_sorted[which(MP_16S_miTags_ss13676_tax_sorted$class_A != "Bolidophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Chlorarachniophyta" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Cryptophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Pelagophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Rappemonad" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Cryptomonadales" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Bacillariophyta" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Dictyochophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Mamiellophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Prasinophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Eustigmatales" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Trebouxiophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Prymnesiophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Dinophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Chrysophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Chlorophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Ulvophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Raphydophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Phaeothamniophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Xanthophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Chlorodendrophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Nephroselmis" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Rhodophyta" & MP_16S_miTags_ss13676_tax_sorted$class_A != "Pavlovophyceae" & MP_16S_miTags_ss13676_tax_sorted$class_A != "other_plastids"),]

MP_16S_miTags_protists <- MP_16S_miTags_ss13676_tax_sorted[which(MP_16S_miTags_ss13676_tax_sorted$class_A != "heterotrophic_bacteria" & MP_16S_miTags_ss13676_tax_sorted$class_A != "cyanobacteria"),]

dim(MP_16S_miTags_protists)
dim(MP_16S_miTags_bacteria)
MP_16S_miTags_protists[1:5,9:12]
MP_16S_miTags_bacteria[1:5,9:12]
```

```{r aggregate_16S_protists, echo=TRUE}
MP_16S_miTags_class_summary_reads_per_class_protists<-aggregate(rowSums(MP_16S_miTags_protists[1:9]), list(MP_16S_miTags_protists$class_A), sum)
# count the different groups

MP_16S_miTags_class_summary_otus_per_class_protists<-aggregate(rowSums(MP_16S_miTags_protists[1:9]), list(MP_16S_miTags_protists$class_A), length)

## READS PER CLASS ##
attach(MP_16S_miTags_class_summary_reads_per_class_protists)
MP_16S_miTags_class_summary_reads_per_class_protists_order<-MP_16S_miTags_class_summary_reads_per_class_protists[order(-x),]
detach(MP_16S_miTags_class_summary_reads_per_class_protists)
MP_16S_miTags_class_summary_reads_per_class_protists_order

#fix column names
row.names(MP_16S_miTags_class_summary_reads_per_class_protists_order)<-MP_16S_miTags_class_summary_reads_per_class_protists_order[,1]
MP_16S_miTags_class_summary_reads_per_class_protists_order<-MP_16S_miTags_class_summary_reads_per_class_protists_order[c(-1)]
colnames(MP_16S_miTags_class_summary_reads_per_class_protists_order)<-c("reads_per_class")


## OTUs PER CLASS ##
attach(MP_16S_miTags_class_summary_otus_per_class_protists)
MP_16S_miTags_class_summary_otus_per_class_protists_order<-MP_16S_miTags_class_summary_otus_per_class_protists[order(-x),]
detach(MP_16S_miTags_class_summary_otus_per_class_protists)
MP_16S_miTags_class_summary_otus_per_class_protists_order

row.names(MP_16S_miTags_class_summary_otus_per_class_protists_order)<-MP_16S_miTags_class_summary_otus_per_class_protists_order[,1]
MP_16S_miTags_class_summary_otus_per_class_protists_order<-MP_16S_miTags_class_summary_otus_per_class_protists_order[c(-1)]
colnames(MP_16S_miTags_class_summary_otus_per_class_protists_order)<-c("OTUs_per_class")
```

```{r aggregate_16S_bacteria, echo=TRUE}
MP_16S_miTags_class_summary_reads_per_class_bacteria<-aggregate(rowSums(MP_16S_miTags_bacteria[1:9]), list(MP_16S_miTags_bacteria$class_A), sum)
# count the different groups

MP_16S_miTags_class_summary_otus_per_class_bacteria<-aggregate(rowSums(MP_16S_miTags_bacteria[1:9]), list(MP_16S_miTags_bacteria$class_A), length)

## READS PER CLASS ##
attach(MP_16S_miTags_class_summary_reads_per_class_bacteria)
MP_16S_miTags_class_summary_reads_per_class_bacteria_order<-MP_16S_miTags_class_summary_reads_per_class_bacteria[order(-x),]
detach(MP_16S_miTags_class_summary_reads_per_class_bacteria)
MP_16S_miTags_class_summary_reads_per_class_bacteria_order

#fix column names
row.names(MP_16S_miTags_class_summary_reads_per_class_bacteria_order)<-MP_16S_miTags_class_summary_reads_per_class_bacteria_order[,1]
MP_16S_miTags_class_summary_reads_per_class_bacteria_order<-MP_16S_miTags_class_summary_reads_per_class_bacteria_order[c(-1)]
colnames(MP_16S_miTags_class_summary_reads_per_class_bacteria_order)<-c("reads_per_class")


## OTUs PER CLASS ##
attach(MP_16S_miTags_class_summary_otus_per_class_bacteria)
MP_16S_miTags_class_summary_otus_per_class_bacteria_order<-MP_16S_miTags_class_summary_otus_per_class_bacteria[order(-x),]
detach(MP_16S_miTags_class_summary_otus_per_class_bacteria)
MP_16S_miTags_class_summary_otus_per_class_bacteria_order

row.names(MP_16S_miTags_class_summary_otus_per_class_bacteria_order)<-MP_16S_miTags_class_summary_otus_per_class_bacteria_order[,1]
MP_16S_miTags_class_summary_otus_per_class_bacteria_order<-MP_16S_miTags_class_summary_otus_per_class_bacteria_order[c(-1)]
colnames(MP_16S_miTags_class_summary_otus_per_class_bacteria_order)<-c("OTUs_per_class")
```

```{r count_samples_per_group_16S_protists, echo=FALSE}
#create a table per group and count in how many samples they occur. 
Dinophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Dinophyceae"),]
Dinophyceae_tb[1:5,1:5]
Dinophyceae_tb_occur <- Dinophyceae_tb[,1:9]
Dinophyceae_tb_occur[1:5,1:5]
dim(Dinophyceae_tb_occur)

length(Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0])
#Dinophyceae_tb_samples <- Dinophyceae_tb_occur[,colSums(Dinophyceae_tb_occur) > 0]
#length(Dinophyceae_tb_samples[which(colSums(Dinophyceae_tb_occur) != 0)])

Prasinophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Prasinophyceae"),]
Prasinophyceae_tb_occur <- Prasinophyceae_tb[,1:9]
length(Prasinophyceae_tb_occur[,colSums(Prasinophyceae_tb_occur) > 0])

Chrysophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Chrysophyceae"),]
Chrysophyceae_tb_occur <- Chrysophyceae_tb[,1:9]
length(Chrysophyceae_tb_occur[,colSums(Chrysophyceae_tb_occur) > 0])

Pelagophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Pelagophyceae"),]
Pelagophyceae_tb_occur <- Pelagophyceae_tb[,1:9]
length(Pelagophyceae_tb_occur[,colSums(Pelagophyceae_tb_occur) > 0])

Dictyochophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Dictyochophyceae"),]
Dictyochophyceae_tb_occur <- Dictyochophyceae_tb[,1:9]
length(Dictyochophyceae_tb_occur[,colSums(Dictyochophyceae_tb_occur) > 0])

Cryptomonadales_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Cryptomonadales"),]
Cryptomonadales_tb_occur <- Cryptomonadales_tb[,1:9]
length(Cryptomonadales_tb_occur[,colSums(Cryptomonadales_tb_occur) > 0])

Bacillariophyta_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Bacillariophyta"),]
Bacillariophyta_tb_occur <- Bacillariophyta_tb[,1:9]
length(Bacillariophyta_tb_occur[,colSums(Bacillariophyta_tb_occur) > 0])

Chlorarachniophyta_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Chlorarachniophyta"),]
Chlorarachniophyta_tb_occur <- Chlorarachniophyta_tb[,1:9]
length(Chlorarachniophyta_tb_occur[,colSums(Chlorarachniophyta_tb_occur) > 0])

Bolidophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Bolidophyceae"),]
Bolidophyceae_tb_occur <- Bolidophyceae_tb[,1:9]
length(Bolidophyceae_tb_occur[,colSums(Bolidophyceae_tb_occur) > 0])

Pinguiochysidales_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Pinguiochysidales"),]
Pinguiochysidales_tb_occur <- Pinguiochysidales_tb[,1:9]
length(Pinguiochysidales_tb_occur[,colSums(Pinguiochysidales_tb_occur) > 0])

Prymnesiophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Prymnesiophyceae"),]
Prymnesiophyceae_tb_occur <- Prymnesiophyceae_tb[,1:9]
length(Prymnesiophyceae_tb_occur[,colSums(Prymnesiophyceae_tb_occur) > 0])

Mamiellophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Mamiellophyceae"),]
Mamiellophyceae_tb_occur <- Mamiellophyceae_tb[,1:9]
length(Mamiellophyceae_tb_occur[,colSums(Mamiellophyceae_tb_occur) > 0])

Eustigmatales_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Eustigmatales"),]
Eustigmatales_tb_occur <- Eustigmatales_tb[,1:9]
length(Eustigmatales_tb_occur[,colSums(Eustigmatales_tb_occur) > 0])

Chlorophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Chlorophyceae"),]
Chlorophyceae_tb_occur <- Chlorophyceae_tb[,1:9]
length(Chlorophyceae_tb_occur[,colSums(Chlorophyceae_tb_occur) > 0])

Ulvophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Ulvophyceae"),]
Ulvophyceae_tb_occur <- Ulvophyceae_tb[,1:9]
length(Ulvophyceae_tb_occur[,colSums(Ulvophyceae_tb_occur) > 0])

Raphydophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Raphydophyceae"),]
Raphydophyceae_tb_occur <- Raphydophyceae_tb[,1:9]
length(Raphydophyceae_tb_occur[,colSums(Raphydophyceae_tb_occur) > 0])

Trebouxiophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Trebouxiophyceae"),]
Trebouxiophyceae_tb_occur <- Trebouxiophyceae_tb[,1:9]
length(Trebouxiophyceae_tb_occur[,colSums(Trebouxiophyceae_tb_occur) > 0])

Phaeothamniophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Phaeothamniophyceae"),]
Phaeothamniophyceae_tb_occur <- Phaeothamniophyceae_tb[,1:9]
length(Phaeothamniophyceae_tb_occur[,colSums(Phaeothamniophyceae_tb_occur) > 0])

Xanthophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Xanthophyceae"),]
Xanthophyceae_tb_occur <- Xanthophyceae_tb[,1:9]
length(Xanthophyceae_tb_occur[,colSums(Xanthophyceae_tb_occur) > 0])

Chlorodendrophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Chlorodendrophyceae"),]
Chlorodendrophyceae_tb_occur <- Chlorodendrophyceae_tb[,1:9]
length(Chlorodendrophyceae_tb_occur[,colSums(Chlorodendrophyceae_tb_occur) > 0])

Nephroselmis_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Nephroselmis"),]
Nephroselmis_tb_occur <- Nephroselmis_tb[,1:9]
length(Nephroselmis_tb_occur[,colSums(Nephroselmis_tb_occur) > 0])

Rhodophyta_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Rhodophyta"),]
Rhodophyta_tb_occur <- Rhodophyta_tb[,1:9]
length(Rhodophyta_tb_occur[,colSums(Rhodophyta_tb_occur) > 0])

Pavlovophyceae_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Pavlovophyceae"),]
Pavlovophyceae_tb_occur <- Pavlovophyceae_tb[,1:9]
length(Pavlovophyceae_tb_occur[,colSums(Pavlovophyceae_tb_occur) > 0])

Rappemonad_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "Rappemonad"),]
Rappemonad_tb_occur <- Rappemonad_tb[,1:9]
length(Rappemonad_tb_occur[,colSums(Rappemonad_tb_occur) > 0])

other_plastids_tb <- MP_16S_miTags_protists[which(MP_16S_miTags_protists$class_A == "other_plastids"),]
other_plastids_tb_occur <- other_plastids_tb[,1:9]
length(other_plastids_tb_occur[,colSums(other_plastids_tb_occur) > 0])
```

[BACTERIA]

```{r count_samples_per_group_16S_bacteria, echo=FALSE}
#create a table per group and count in how many samples they occur. 
heterotrophic_bacteria_tb <- MP_16S_miTags_bacteria[which(MP_16S_miTags_bacteria$class_A == "heterotrophic_bacteria"),]
heterotrophic_bacteria_tb_occur <- heterotrophic_bacteria_tb[,1:9]
length(heterotrophic_bacteria_tb_occur[,colSums(heterotrophic_bacteria_tb_occur) > 0])

cyanob_tb <- MP_16S_miTags_bacteria[which(MP_16S_miTags_bacteria$class_A == "cyanobacteria"),]
cyanob_tb_occur <- cyanob_tb[,1:9]
length(cyanob_tb_occur[,colSums(cyanob_tb_occur) > 0])
```


```{r create_table_merging_#OTUs_#reads_#samples_protists, echo=FALSE}
MP_16S_miTags_protists_OTUs_reads_samples <- merge(MP_16S_miTags_class_summary_reads_per_class_protists_order, MP_16S_miTags_class_summary_otus_per_class_protists_order, by="row.names")

row.names(MP_16S_miTags_protists_OTUs_reads_samples)<-MP_16S_miTags_protists_OTUs_reads_samples[,1]
MP_16S_miTags_protists_OTUs_reads_samples<-MP_16S_miTags_protists_OTUs_reads_samples[,-1]
colnames(MP_16S_miTags_protists_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
MP_16S_miTags_protists_OTUs_reads_samples[1:5,1:2]

MP_16S_miTags_protists_OTUs_reads_samples<-MP_16S_miTags_protists_OTUs_reads_samples[order(MP_16S_miTags_protists_OTUs_reads_samples$reads_per_class, MP_16S_miTags_protists_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
MP_16S_miTags_protists_OTUs_reads_samples["samples_per_class"] <- NA
MP_16S_miTags_protists_OTUs_reads_samples$samples_per_class<-c(9,5,8,3,1)

#table without "other_plastids"
MP_16S_miTags_protists_OTUs_reads_samples_without_op <- MP_16S_miTags_protists_OTUs_reads_samples[-1,]
MP_16S_miTags_protists_OTUs_reads_samples_without_op
```

```{r create_table_merging_#OTUs_#reads_#samples_bacteria, echo=FALSE}
MP_16S_miTags_bacteria_OTUs_reads_samples <- merge(MP_16S_miTags_class_summary_reads_per_class_bacteria_order, MP_16S_miTags_class_summary_otus_per_class_bacteria_order, by="row.names")

row.names(MP_16S_miTags_bacteria_OTUs_reads_samples)<-MP_16S_miTags_bacteria_OTUs_reads_samples[,1]
MP_16S_miTags_bacteria_OTUs_reads_samples<-MP_16S_miTags_bacteria_OTUs_reads_samples[,-1]
colnames(MP_16S_miTags_bacteria_OTUs_reads_samples)<-c("reads_per_class","OTUs_per_class")
MP_16S_miTags_bacteria_OTUs_reads_samples[1:5,1:2]

MP_16S_miTags_bacteria_OTUs_reads_samples<-MP_16S_miTags_bacteria_OTUs_reads_samples[order(MP_16S_miTags_bacteria_OTUs_reads_samples$reads_per_class, MP_16S_miTags_bacteria_OTUs_reads_samples$OTUs_per_class, decreasing = T), c(1,2)]

#add no. of samples per class
MP_16S_miTags_bacteria_OTUs_reads_samples["samples_per_class"] <- NA
MP_16S_miTags_bacteria_OTUs_reads_samples$samples_per_class<-c(9,9)

#table without "heterotropic_bacteria"
MP_16S_miTags_bacteria_OTUs_reads_samples_without_hb <- MP_16S_miTags_bacteria_OTUs_reads_samples[-1,]
MP_16S_miTags_bacteria_OTUs_reads_samples_without_hb
```


[[PROTISTS]]
[PLOT DESCRIPTION]

```{r stations_vs_OTUs_protists, echo=FALSE}
otus_vs_samples_plot_protists <- ggplot(MP_16S_miTags_protists_OTUs_reads_samples_without_op, aes(samples_per_class,OTUs_per_class, label = rownames(MP_16S_miTags_protists_OTUs_reads_samples_without_op)))
otus_vs_samples_plot_protists + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 11)) + labs(title="[tb5] #samples vs #OTUs - protists", x="samples", y="OTUs")
```

[PLOT DESCRIPTION]

```{r OTUs_vs_reads_protists, echo=FALSE}
otus_vs_samples_plot_protists <- ggplot(MP_16S_miTags_protists_OTUs_reads_samples_without_op, aes(OTUs_per_class, reads_per_class, label = rownames(MP_16S_miTags_protists_OTUs_reads_samples_without_op)))
otus_vs_samples_plot_protists + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05,  size = 3) + scale_x_continuous(limits = c(0, 20)) + labs(title="[tb6] #OTUs vs #reads - protists", x="OTUs", y="reads")
```

[[PROTISTS vs BACTERIA]]
[PLOT DESCRIPTION]

```{r merge_protists_and_bacteria_tb, echo=FALSE}
tb16S_protists_plus_bacteria_OTUs_reads_samples<-rbind(MP_16S_miTags_protists_OTUs_reads_samples_without_op,MP_16S_miTags_bacteria_OTUs_reads_samples_without_hb)
```

```{r stations_vs_OTUs_protists_plus_bacteria, echo=FALSE}
otus_vs_samples_plot_protists_plus_bacteria <- ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples, aes(samples_per_class,OTUs_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples)))
otus_vs_samples_plot_protists_plus_bacteria + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_y = 0.05, size = 3) + scale_x_continuous(limits = c(0, 11)) + labs(title="[tb9] #samples vs #OTUs - protists & bacteria", x="samples", y="OTUs") + coord_trans(y = "log10")
```

[PLOT DESCRIPTION]

```{r OTUs_vs_reads_protists_plus_bacteria, echo=FALSE}
otus_vs_samples_plot_protists_plus_bacteria <- ggplot(tb16S_protists_plus_bacteria_OTUs_reads_samples, aes(OTUs_per_class, reads_per_class, label = rownames(tb16S_protists_plus_bacteria_OTUs_reads_samples)))
otus_vs_samples_plot_protists_plus_bacteria + geom_point() + geom_text(check_overlap = TRUE,hjust = -0.05, nudge_x = 0.05, size=3) + scale_x_continuous(limits = c(0, 320)) + labs(title="[tb10] #OTUs vs #reads - protists & bacteria", x="OTUs", y="reads") + coord_trans(y = "log10")
```